public/app/features/datasources/extensions/getDataSourceExtensionConfigs.test.tsx

```diff
@@ -0,0 +1,274 @@
+      const statusExtensions = configs.filter((config) =>
+        config.targets?.includes(PluginExtensionPoints.DataSourceConfigStatus)
+      );
+
+      expect(actionExtensions.length).toBeGreaterThan(0);
+      expect(statusExtensions.length).toBeGreaterThan(0);
+    });
+
+    it('should have proper translation comments for all text', () => {
+      const configs = getDataSourceExtensionConfigs();
+
+      configs.forEach((config) => {
+        // All configurations should follow the pattern of not using translations
+        // as documented in the specs (cannot use t() at top level)
+        expect(config.title).toBeDefined();
+        expect(config.description).toBeDefined();
+        expect(config.category).toBeDefined();
+      });
+    });
+
+    it('should have valid icons for all extensions', () => {
+      const configs = getDataSourceExtensionConfigs();
+
+      configs.forEach((config) => {
+        expect(config.icon).toBeDefined();
+        expect(typeof config.icon).toBe('string');
+      });
+    });
+
+    it('should have either path or onClick for all extensions', () => {
+      const configs = getDataSourceExtensionConfigs();
+
+      configs.forEach((config) => {
+        expect(config.path !== undefined || config.onClick !== undefined).toBe(true);
+      });
+    });
+  });
+
+  describe('Context Type Safety', () => {
+    it('should handle missing context gracefully in configure functions', () => {
+      const configs = getDataSourceExtensionConfigs();
+
+      configs.forEach((config) => {
+        if (config.configure) {
+          // Test with undefined context
+          const result = config.configure(undefined);
+          expect(result).toBeUndefined();
+
+          // Test with missing dataSource
+          const resultNoDS = config.configure({ dataSource: undefined } as unknown as
+            | PluginExtensionDataSourceConfigActionsContext
+            | PluginExtensionDataSourceConfigStatusContext);
+          expect(resultNoDS).toBeUndefined();
+
+          // Test with missing datasource type
+          const resultNoType = config.configure({
+            dataSource: { uid: 'test', name: 'test' },
+          } as unknown as PluginExtensionDataSourceConfigActionsContext | PluginExtensionDataSourceConfigStatusContext);
+          expect(resultNoType).toBeUndefined();
+        }
+      });
+    });
+
+    it('should validate context properties used in configure functions', () => {
+      const configs = getDataSourceExtensionConfigs();
+
+      const monitoringConfig = configs.find((config) => config.title?.includes('Monitoring Tool'));
+
+      const troubleshootingConfig = configs.find((config) => config.title?.includes('Troubleshooting Guide'));
+
+      // Monitoring config should check dataSource.type
+      expect(monitoringConfig?.configure).toBeDefined();
+
+      // Troubleshooting config should check severity
+      expect(troubleshootingConfig?.configure).toBeDefined();
+    });
+  });
+});
```

Comment:
Hmm, what's the goal of these tests? The `getDataSourceExtensionConfigs()` function is mostly just returning a configuration data object (with some minimal business logic). By defining the config values twice (once in the tests and once in the config object), the tests will break every time the config values change, making them more like an overhead in my opinion.